### ISSUE TITLE
support for multiple applications

### ISSUE THREAD

**Issue by GoogleCodeExporter** [ntrees/javapns#20]
```
Whether the PushNotificationManager supports to handle multiple apps. If 
it is a singleton class i can have only one instance. So whenever i need 
to send message, i have to open connection because my application may vary 
for each message. But apple suggests that we should not open/close 
connections very often.

How can i maintain multiple connections ( for multiple applications) with 
pushnotificationmanager? can you please help me out on this ASAP?

```

Original issue reported on code.google.com by `paulraj....@gmail.com` on 21 May 2010 at 7:18

**Comment by GoogleCodeExporter** [comment]
```
I'm confused.

You have 1 java app, that sends notification for several iPhone apps? (each of 
which has it's own certificate etc...)

If this is the case, then the only thing I can see is reset the javapns 
instance when you want to switch certificates.

What do you mean by 'maintain connections' ? In theory you open a connection, 
send your notices, close your 
connection, repeat...
```

Original comment by `idbill.p...@gmail.com` on 21 May 2010 at 3:34

**Comment by GoogleCodeExporter** [comment]
```
Take a scenario i have two list. One list is associated with one application. 
Second 
list associated with another application. I will run two threads at the same 
time 
once i open the connection for the first list, i want to keep open the 
connection 
till i complete all the messages in the first list.(because apple is not 
recommending to open/close connections very often). when i process the first 
list, 
another thread start processing the second list which will reset the connection 
with 
second application since the pushnotification manager is a singleton class.

How do i proceed with javapns in such requirement? 
```

Original comment by `paulraj....@gmail.com` on 31 May 2010 at 5:34

**Comment by GoogleCodeExporter** [comment]
```
basically javapns has to provide support for multiple applications.
```

Original comment by `paulraj....@gmail.com` on 3 Jun 2010 at 7:34

**Comment by GoogleCodeExporter** [comment]
```
Would changing the library from a singleton to a instance be better?

Or do you have another suggestion?

I can work on this, but it will take some time. Any assistance would be greatly 
appreciated.
```

Original comment by `idbill.p...@gmail.com` on 3 Jun 2010 at 3:51
- Changed state: **Accepted**

**Comment by GoogleCodeExporter** [comment]
```
[deleted comment]
```

**Comment by GoogleCodeExporter** [comment]
```
[deleted comment]
```

**Comment by GoogleCodeExporter** [comment]
```
Hi i have the same thing to do and have an question, can i open a connection 
and send
several apns, for this app, close connection and reopen a new to send several 
apns
for the next app.
shorter do i have to create and close conection for any single apn
or only for each aplication?

thx Felix

```

Original comment by `honegger...@gmail.com` on 4 Jun 2010 at 11:09

**Comment by GoogleCodeExporter** [comment]
```
Thanks, I will try to provide some inputs from my side. 
```

Original comment by `paulraj....@gmail.com` on 7 Jun 2010 at 9:41

**Comment by GoogleCodeExporter** [comment]
```
Yes Felix, You can do that untill you have one thread to process all the 
messages 
(process the messages one by one). Only thing is you need to have the list 
ordered 
by the application and process it sequentially. It will be slower when you have 
want 
to process it for millions.
```

Original comment by `paulraj....@gmail.com` on 7 Jun 2010 at 9:42

**Comment by GoogleCodeExporter** [comment]
```
thx Paulraj for the fast answer,
I have one more question, support the javapns threading, i dont see somthing 
for it
only a SSLsocket, how maybe can do it. 
I have to send masses of apns to several applications, Devices an whane do it 
with
more then 1 connection. But i dont finde a way to open more then 1 connection.

```

Original comment by `honegger...@gmail.com` on 7 Jun 2010 at 10:20

**Comment by GoogleCodeExporter** [comment]
```
> Would changing the library from a singleton to a instance be better?
....
> Any assistance would be greatly appreciated.

Yes i whane help you i need it very fast to open more than 1 connection if you 
tell
me what i can do to help you would be fine.

mfg Felix
```

Original comment by `honegger...@gmail.com` on 7 Jun 2010 at 11:14

**Comment by GoogleCodeExporter** [comment]
```
Ok i work on it 
im nearly done to make instance or singletons of the class
and make some minor changes that optimice some things.
and adding comments to them.
i will send it to you if im done.
```

Original comment by `honegger...@gmail.com` on 9 Jun 2010 at 11:26

**Comment by GoogleCodeExporter** [comment]
```
How is this going?
```

Original comment by `idbill.p...@gmail.com` on 16 Oct 2010 at 9:38

**Comment by GoogleCodeExporter** [comment]
```
I'm also interested in this feature.  I have multiple apps driven by a single 
server cluster.  It would be really great to have multiple application support. 


```

Original comment by `ayyoi...@gmail.com` on 6 Mar 2011 at 9:12

**Comment by GoogleCodeExporter** [comment]
```
Also interested to know when multiple applications will be supported. thx Chris
```

Original comment by `ch...@hullomail.com` on 2 Aug 2011 at 2:32

**Comment by GoogleCodeExporter** [comment]
```
Please see version 2.0 (branch javapns2), which was completely re-architected 
to support multiple applications, persistable devices, etc.  

I believe we can close this issue as fixed in version 2.0...
```

Original comment by `sype...@gmail.com` on 31 Aug 2011 at 6:36

**Comment by GoogleCodeExporter** [comment]
```
Design issue addressed in javapns 2.0.   Closing.
```

Original comment by `sype...@gmail.com` on 7 Sep 2011 at 1:01
- Changed state: **Fixed**
